src/modules/RA/DataLayer/components/DataLayerForm.js

```diff
@@ -19,6 +19,9 @@ const DataLayerForm = ({ ...props }) => {
   const [miniSheetInError, setMiniSheetInError] = React.useState(false);
   const [tableInError, setTableInError] = React.useState(false);
   const [filterInError, setFilterInError] = React.useState(false);
+  const [legendInError, setLegendInError] = React.useState(false);
+  const [styleInError, setStyleInError] = React.useState(false);
+  const [definitionInError, setDefinitionInError] = React.useState(false);
```

Comment:
To manage all theses errors states, maybe you could refacto all theses useState by an useReducer hook ?

Reply:
Seems indeed a good use case.

Reply:
Done.